packages/npm/@amazeelabs/gatsby-source-silverback/src/gatsby-node.ts

```diff
@@ -68,88 +68,90 @@ export const sourceNodes: GatsbyNode['sourceNodes'] = async (
     return;
   }
 
-  const executor = createQueryExecutor({
-    ...options,
-    headers: options.drupal_external_url
-      ? getForwardedHeaders(new URL(options.drupal_external_url))
-      : undefined,
-  });
-  // TODO: Accept configuration and fragment overrides from plugin settings.
-  const config = await createSourcingConfig(gatsbyApi, executor, options);
-  const context = createSourcingContext(config);
+  if (options.drupal_url) {
```

Comment:
Both `drupal_url` and `graphql_path` are optional now. But here and following we check only `drupal_url`. Yet, there will be runtime errors if `graphql_path` isn't provided.

We could use advanced [Joi](https://stackoverflow.com/questions/56423558/joi-validation-how-to-require-or-optional-field-based-on-another-key-exists-in)+TS techniques. Or just adjust `if` conditions to check for both.

I would do the latter. The simpler the better.